File: Projeto_WindowsForms/DAL/FolhaPagamentoDAO.cs
using Modelo;
using Modelo.Enum;
using System.Data.SqlClient;

namespace DAL
{
    public class FolhaPagamentoDAO : BaseDAO
    {
        public void CadastrarFolhaPagamento(FolhaPagamento folhaPagamento)
        {
            var cmd = new SqlCommand
            {
                CommandText = @"insert into folha_pagamento (inss, irrf, horas_extras, valor_horas_extras, valor_liquido, descontos_totais, vencimentos_totais, aliquota_inss, aliquota_irrf, data_criacao, ativo, id_colaborador)
                                values (@inss, @irrf, @horas_extras, @valor_horas_extras, @valor_liquido, @descontos_totais, @vencimentos_totais, @aliquota_inss, @aliquota_irrf, @data_criacao, @ativo, @id_colaborador)"
            };

            cmd.Parameters.AddWithValue("@inss", folhaPagamento.Inss);
            cmd.Parameters.AddWithValue("@irrf", folhaPagamento.Irrf);
            cmd.Parameters.AddWithValue("@horas_extras", folhaPagamento.HorasExtras);
            cmd.Parameters.AddWithValue("@valor_horas_extras", folhaPagamento.ValorHorasExtras);
            cmd.Parameters.AddWithValue("@valor_liquido", folhaPagamento.ValorLiquido);
            cmd.Parameters.AddWithValue("@descontos_totais", folhaPagamento.DescontosTotais);
            cmd.Parameters.AddWithValue("@vencimentos_totais", folhaPagamento.VencimentosTotais);
            cmd.Parameters.AddWithValue("@aliquota_inss", folhaPagamento.AliquotaInss);
            cmd.Parameters.AddWithValue("@aliquota_irrf", folhaPagamento.AliquotaIrrf);
            cmd.Parameters.AddWithValue("@data_criacao", folhaPagamento.DataCriacao);
            cmd.Parameters.AddWithValue("@ativo", folhaPagamento.Ativo);
            cmd.Parameters.AddWithValue("@id_colaborador", folhaPagamento.Colaborador.Id);

            try
            {
                cmd.Connection = conexao.Conectar();

                cmd.ExecuteNonQuery();

                conexao.Desconectar();
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                conexao.Desconectar();
            }

        }

        public List<FolhaPagamento> ListarFolhaPagamento()
        {
            var cmd = new SqlCommand
            {
                CommandText = @"SELECT f.id, f.inss, f.irrf, f.horas_extras, f.valor_horas_extras, f.valor_liquido, f.descontos_totais, f.vencimentos_totais, f.aliquota_inss, f.aliquota_irrf, f.data_criacao, f.ativo, c.id as id_colaborador, c.nome_completo, c.sexo, c.cargo, c.salario, c.data_admissao, c.ativo as c_ativo, c.id_empresa
                                FROM folha_pagamento f
                                    INNER JOIN colaborador c
                                    ON f.id_colaborador = c.id
                                WHERE
                                    f.ativo = 1 AND
                                    c.ativo = 1"
            };

            var listaFolhaPagamento = new List<FolhaPagamento>();

            try
            {
                cmd.Connection = conexao.Conectar();

                dr = cmd.ExecuteReader();

                if (dr.HasRows)
                {
                    while (dr.Read())
                    {
                        Enum.TryParse<TipoCargo>(dr["cargo"].ToString(), out var cargo);

                        var folhaPagamento = new FolhaPagamento
                        {
                            Id = int.Parse(dr["id"].ToString()),
                            Inss = decimal.Parse(dr["inss"].ToString()),
                            Irrf = decimal.Parse(dr["irrf"].ToString()),
                            HorasExtras = int.Parse(dr["horas_extras"].ToString()),
                            ValorHorasExtras = decimal.Parse(dr["valor_horas_extras"].ToString()),
                            ValorLiquido = decimal.Parse(dr["valor_liquido"].ToString()),
                            DescontosTotais = decimal.Parse(dr["descontos_totais"].ToString()),
                            VencimentosTotais = decimal.Parse(dr["vencimentos_totais"].ToString()),
                            AliquotaInss = decimal.Parse(dr["aliquota_inss"].ToString()),
                            AliquotaIrrf = decimal.Parse(dr["aliquota_irrf"].ToString()),
                            DataCriacao = DateTime.Parse(dr["data_criacao"].ToString()),
                            Ativo = bool.Parse(dr["ativo"].ToString()),
                            Colaborador = new Colaborador
                            {
                                Id = int.Parse(dr["id_colaborador"].ToString()),
                                NomeCompleto = dr["nome_completo"].ToString(),
                                Sexo = (TipoSexo)char.Parse(dr["sexo"].ToString()),
                                Cargo = cargo,
                                Salario = decimal.Parse(dr["salario"].ToString()),
                                DataAdmissao = DateTime.Parse(dr["data_admissao"].ToString()),
                                Ativo = bool.Parse(dr["c_ativo"].ToString()),
                                Empresa = new Empresa
                                {
                                    Id = int.Parse(dr["id_empresa"].ToString()),
                                }
                            }
                        };

                        listaFolhaPagamento.Add(folhaPagamento);
                    }
                }
                dr.Close();
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                conexao.Desconectar();
            }

            return listaFolhaPagamento;
        }

        public FolhaPagamento BuscarPorColaborador(Colaborador colaborador)
        {
            var cmd = new SqlCommand
            {
                CommandText = @"SELECT f.id, f.inss, f.irrf, f.horas_extras, f.valor_horas_extras, f.valor_liquido, f.descontos_totais, f.vencimentos_totais, f.aliquota_inss, f.aliquota_irrf, f.data_criacao, f.ativo, c.id as id_colaborador, c.nome_completo, c.sexo, c.cargo, c.salario, c.data_admissao, c.ativo as c_ativo, c.id_empresa
                                FROM folha_pagamento f
                                    INNER JOIN colaborador c
                                    ON f.id_colaborador = c.id
                                WHERE
                                    f.ativo = 1 AND
                                    c.ativo = 1 AND
                                    c.id = @id_colaborador AND  
                                    YEAR(f.data_criacao) = @ano AND MONTH(f.data_criacao) = @mes"
            };

            cmd.Parameters.AddWithValue("@id_colaborador", colaborador.Id);
            cmd.Parameters.AddWithValue("@ano", DateTime.Now.Year);
            cmd.Parameters.AddWithValue("@mes", DateTime.Now.Month);

            try
            {
                cmd.Connection = conexao.Conectar();

                dr = cmd.ExecuteReader();

                if (dr.Read())
                {
                    Enum.TryParse<TipoCargo>(dr["cargo"].ToString(), out var cargo);

                    var folhaPagamento = new FolhaPagamento
                    {
                        Id = int.Parse(dr["id"].ToString()),
                        Inss = decimal.Parse(dr["inss"].ToString()),
                        Irrf = decimal.Parse(dr["irrf"].ToString()),
                        HorasExtras = int.Parse(dr["horas_extras"].ToString()),
                        ValorHorasExtras = decimal.Parse(dr["valor_horas_extras"].ToString()),
                        ValorLiquido = decimal.Parse(dr["valor_liquido"].ToString()),
                        DescontosTotais = decimal.Parse(dr["descontos_totais"].ToString()),
                        VencimentosTotais = decimal.Parse(dr["vencimentos_totais"].ToString()),
                        AliquotaInss = decimal.Parse(dr["aliquota_inss"].ToString()),
                        AliquotaIrrf = decimal.Parse(dr["aliquota_irrf"].ToString()),
                        DataCriacao = DateTime.Parse(dr["data_criacao"].ToString()),
                        Ativo = bool.Parse(dr["ativo"].ToString()),
                        Colaborador = new Colaborador
                        {
                            Id = int.Parse(dr["id_colaborador"].ToString()),
                            NomeCompleto = dr["nome_completo"].ToString(),
                            Sexo = (TipoSexo)char.Parse(dr["sexo"].ToString()),
                            Cargo = cargo,
                            Salario = decimal.Parse(dr["salario"].ToString()),
                            DataAdmissao = DateTime.Parse(dr["data_admissao"].ToString()),
                            Ativo = bool.Parse(dr["c_ativo"].ToString()),
                            Empresa = new Empresa
                            {
                                Id = int.Parse(dr["id_empresa"].ToString()),
                            }
                        }
                    };

                    return folhaPagamento;
                }

                dr.Close();
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                conexao.Desconectar();
            }

            return null;
        }
    }
}

File: Projeto_Android_Studio/MyApplication/app/src/main/java/com/example/myapplication/DAL/FolhaPagamentoDAO.java
package com.example.myapplication.DAL;

import com.example.myapplication.modelo.Colaborador;
import com.example.myapplication.modelo.Empresa;
import com.example.myapplication.modelo.FolhaPagamento;
import com.example.myapplication.modelo.enums.TipoCargo;
import com.example.myapplication.modelo.enums.TipoSexo;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.ArrayList;

public class FolhaPagamentoDAO {
    Connection connection;

    public ArrayList<FolhaPagamento> listarFolhaPagamento() {

        ArrayList<FolhaPagamento> listaFolhaPagamento  = new ArrayList<>();

        try
        {
            Conexao conexao = new Conexao();
            connection = conexao.Conectar();

            if (connection != null) {
                String query = "SELECT f.id, f.inss, f.irrf, f.horas_extras, f.valor_horas_extras, f.valor_liquido, f.descontos_totais, f.vencimentos_totais, f.aliquota_inss, f.aliquota_irrf, f.data_criacao, f.ativo, c.id as id_colaborador, c.nome_completo, c.sexo, c.cargo, c.salario, c.data_admissao, c.ativo as c_ativo, c.id_empresa FROM folha_pagamento f INNER JOIN colaborador c ON f.id_colaborador = c.id WHERE f.ativo = 1 AND c.ativo = 1";
                Statement st = connection.createStatement();

                ResultSet rs = st.executeQuery(query);

                while (rs.next()) {
                    FolhaPagamento folhaPagamento = new FolhaPagamento();

                    folhaPagamento.Id = rs.getInt("id");
                    folhaPagamento.Inss = rs.getDouble("inss");
                    folhaPagamento.Irrf = rs.getDouble("irrf");
                    folhaPagamento.HorasExtras = rs.getInt("horas_extras");
                    folhaPagamento.ValorHorasExtras = rs.getDouble("valor_horas_extras");
                    folhaPagamento.ValorLiquido = rs.getDouble("valor_liquido");
                    folhaPagamento.DescontosTotais = rs.getDouble("descontos_totais");
                    folhaPagamento.VencimentosTotais = rs.getDouble("vencimentos_totais");
                    folhaPagamento.AliquotaInss = rs.getDouble("aliquota_inss");
                    folhaPagamento.AliquotaIrrf = rs.getDouble("aliquota_irrf");
                    folhaPagamento.DataCriacao = rs.getString("data_criacao");
                    folhaPagamento.Ativo = Boolean.parseBoolean(rs.getString("ativo"));

                    folhaPagamento.Colaborador = new Colaborador();
                    folhaPagamento.Colaborador.Id = rs.getInt("id_colaborador");
                    folhaPagamento.Colaborador.NomeCompleto = rs.getString("nome_completo");

                    folhaPagamento.Colaborador.Sexo = TipoSexo.ofCode(rs.getString("sexo").charAt(0));
                    folhaPagamento.Colaborador.Cargo = TipoCargo.values()[Integer.parseInt(rs.getString("cargo"))];

                    folhaPagamento.Colaborador.Salario = Double.parseDouble(rs.getString("salario"));

                    folhaPagamento.Colaborador.DataAdmissao = rs.getString("data_admissao");
                    folhaPagamento.Colaborador.Ativo = Boolean.parseBoolean(rs.getString("c_ativo"));

                    listaFolhaPagamento.add(folhaPagamento);
                }
            }
        }
        catch (Exception ex)
        {
            String el = ex.getMessage();
        }

        return listaFolhaPagamento;
    }
}
